python/cudf/cudf/core/indexing.py

```diff
@@ -344,13 +351,19 @@ def _getitem_tuple_arg(self, arg):
                 return self._downcast_to_series(df, arg)
             return df
         else:
-            df = DataFrame()
-            for i, col in enumerate(columns_df._columns):
-                # need Series() in case a scalar is returned
-                df[i] = Series(col[arg[0]])
+            if hasattr(arg[0], "__len__"):
+                arg = (column.as_column(arg[0]), arg[1])
+
+            if pd.api.types.is_bool_dtype(arg[0]):
+                return columns_df.take(arg[0])
+            else:
+                df = DataFrame()
+                for i, col in enumerate(columns_df._columns):
+                    # need Series() in case a scalar is returned
+                    df[i] = Series(col[arg[0]])
```

Comment:
We're still gathering column by column here, can we not gather all the columns at once?